avm/res/web/site/main.bicep

```diff
@@ -24,8 +24,8 @@ param location string = resourceGroup().location
 ])
 param kind string
 
-@description('Required. The resource ID of the app service plan to use for the site.')
-param serverFarmResourceId string
+@description('Optional. The resource ID of the app service plan to use for the site. Not required for containerized function apps.')
+param serverFarmResourceId string?
```

Comment:
write logic to validate that its not passed only for containerised fuction apps. For all other config its should be required.